File: 05/gemini-3.0-pro/python/solution.py
with open("input.txt") as f:
    input_data = f.read().strip()

parts = input_data.split("\n\n")
ranges_lines = parts[0].splitlines()
ids_lines = parts[1].splitlines()

ranges = []
for line in ranges_lines:
    start, end = map(int, line.split('-'))
    ranges.append((start, end))

part1 = 0
for line in ids_lines:
    id_val = int(line)
    is_fresh = False
    for start, end in ranges:
        if start <= id_val <= end:
            is_fresh = True
            break
    if is_fresh:
        part1 += 1

print(f"Day 05 Part 1: {part1}")

# Part 2
ranges.sort()
merged_ranges = []
if ranges:
    curr_start, curr_end = ranges[0]
    for i in range(1, len(ranges)):
        next_start, next_end = ranges[i]
        if next_start <= curr_end + 1:
            curr_end = max(curr_end, next_end)
        else:
            merged_ranges.append((curr_start, curr_end))
            curr_start, curr_end = next_start, next_end
    merged_ranges.append((curr_start, curr_end))

part2 = 0
for start, end in merged_ranges:
    part2 += (end - start + 1)

print(f"Day 05 Part 2: {part2}")

File: 05/gemini-3.0-pro/typescript/solution.ts
import * as fs from 'fs';

const input = fs.readFileSync('input.txt', 'utf-8').trim();
const parts = input.replace(/\r\n/g, '\n').split('\n\n');

const rangesLines = parts[0].split('\n');
const idsLines = parts[1].split('\n');

const ranges: { start: bigint, end: bigint }[] = [];
for (const line of rangesLines) {
    if (!line) continue;
    const split = line.split('-');
    ranges.push({ start: BigInt(split[0]), end: BigInt(split[1]) });
}

let part1 = 0;
for (const line of idsLines) {
    if (!line) continue;
    const id = BigInt(line);
    let isFresh = false;
    for (const range of ranges) {
        if (id >= range.start && id <= range.end) {
            isFresh = true;
            break;
        }
    }
    if (isFresh) {
        part1++;
    }
}

console.log(`Day 05 Part 1: ${part1}`);

// Part 2
ranges.sort((a, b) => {
    if (a.start < b.start) return -1;
    if (a.start > b.start) return 1;
    return 0;
});

const mergedRanges: { start: bigint, end: bigint }[] = [];

if (ranges.length > 0) {
    let currStart = ranges[0].start;
    let currEnd = ranges[0].end;

    for (let i = 1; i < ranges.length; i++) {
        const nextStart = ranges[i].start;
        const nextEnd = ranges[i].end;

        if (nextStart <= currEnd + 1n) {
            if (nextEnd > currEnd) {
                currEnd = nextEnd;
            }
        } else {
            mergedRanges.push({ start: currStart, end: currEnd });
            currStart = nextStart;
            currEnd = nextEnd;
        }
    }
    mergedRanges.push({ start: currStart, end: currEnd });
}

let part2 = 0n;
for (const range of mergedRanges) {
    part2 += (range.end - range.start + 1n);
}

console.log(`Day 05 Part 2: ${part2}`);
